### PR TITLE
update ncbi-scrub task (version 2.2.1)

### PR DESCRIPTION
**Pull request by cimendes**
Closes <Issue number>

## :hammer_and_wrench: Changes Being Made

- `ncbi_scrub`
  - Update docker container to `us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1`
  - Update task so paired-end reads are processed as interleaved files

## :brain: Context and Rationale

For `ncbi_scrub`, the latest version solves the issue of paired reads not being correctly masked, according to their [documentation ](https://github.com/ncbi/sra-human-scrubber/releases/tag/2.2.1). ~~The reads are still processed individually in the task, so the phenomenon might still persist if we do not change the tasks to first interleave the reads, then process them using ncbi-scrub, and then split them again.~~

I've also altered the ncbi_scrub_pe task to first interleave the reads for processing with HRRT, and then split the resulting file back to forward and reverse read files. 

## :clipboard: Workflow/Task Steps

N/A

### Inputs

N/A

### Outputs

- `read1_human_spots_removed` and `read2_human_spots_removed` were removed (this isn't actually a final output of any of the workflows)
- `human_spots_removed` was added with the stats containing the number of spots removed for the interleaved files

### Locally

Tests passed locally with `miniwdl` for the individual task.

### Terra

Tests under way... 

## :microscope: Quality checks

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [ ] The workflow/task has been tested locally and on Terra
- [ ] The CI/CD has been adjusted and tests are passing
- [x] Everything follows the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-bb456f34322d4f4db699d4029050481c)

### CODE REVIEW

**Review by frankambrosio3 — COMMENTED** [review]
testing wf here: https://app.terra.bio/#workspaces/theiagen-validations/ambrosio_validation_sandbox/job_history/62526fbe-07f0-48a3-afb6-8e8cc418bf0e

File: tasks/quality_control/task_ncbi_scrub.wdl [hunk 1]
```diff
@@ -5,51 +5,29 @@ task ncbi_scrub_pe {
...
-    # unzip file if necessary
-    if [[ "~{read2}" == *.gz ]]
-    then
-      gunzip -c ~{read2} > r2.fastq
-      read2_unzip=r2.fastq
-    else
-      read2_unzip=~{read2}
-    fi
+    # unzip read files as scrub tool does not take in .gz fastq files, and interleave them
+    paste <(zcat ~{read1} | paste - - - -) <(zcat ~{read2} | paste - - - -) | tr '\t' '\n' > interleaved.fastq
```
Comment (kevinlibuit) [inline tasks/quality_control/task_ncbi_scrub.wdl:16]: Will this work if there are uneven reads in `~{read1}` and `~{read2}`?

**Comment by kevinlibuit** [comment]
Seeing failures in workflows utilizing the `task ncbi_scrub_se`. Unclear from GHA log files as to why this is occurring. Also seeing many failures in the Terra runs posted by @frankambrosio3 above

**Comment by sam-baird** [comment]
> Seeing failures in workflows utilizing the `task ncbi_scrub_se`. Unclear from GHA log files as to why this is occurring. Also seeing many failures in the Terra runs posted by @frankambrosio3 above

This error may be occurring in `ncbi_scrub_se` because the `-n` option for `scrub.sh` [was removed](https://github.com/ncbi/sra-human-scrubber/releases) as of 2.0.0 and its functionality made the default. I'm using code from this pull request to add read scrubbing to a workflow, and removing `-n` (and adding `-i`) fixed it for me.

**Comment by cimendes** [comment]
Closing PR as stale

**Comment by cimendes** [comment]
⚠️ Please do not delete the branch! Thank you! ⚠️

**Pull request closed without merging**